tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/HttpContextRefactor/CSharpHttpContextRefactorCodeFixProviderTests.cs

```diff
@@ -0,0 +1,463 @@
+     Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.HttpContextCurrentAnalyzer,
+     Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes.CSharpHttpContextRefactorCodeFixProvider>;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test
+{
+    public class CSharpHttpContextRefactorCodeFixProviderTests
+    {
+        private const string HttpContextName = "HttpContext";
+        private const string HttpContextCurrentName = "HttpContext HttpContext.Current";
+        private const string DiagnosticId = HttpContextCurrentAnalyzer.DiagnosticId;
+
+        [Fact]
+        public async Task EmptyCode()
+        {
+            var test = string.Empty;
+
+            await VerifyCS.Create().WithSource(test).RunAsync();
+        }
+
+        [Fact]
+        public async Task SimpleUse()
+        {
+            var test = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public void Test()
+            {
+                _ = {|#0:HttpContext.Current|};
+            }
+        }
+    }";
+            var fixtest = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public void Test(HttpContext currentContext)
+            {
+                _ = currentContext;
+            }
+        }
+    }";
+
+            var expected = VerifyCS.Diagnostic(DiagnosticId).WithLocation(0).WithArguments(HttpContextName, HttpContextCurrentName);
+            await CreateTest().WithSource(test).WithFixed(fixtest).WithExpectedDiagnostics(expected).RunAsync();
+        }
+
+        [Fact]
+        public async Task ExpressionBody()
+        {
+            var test = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public object Test() => {|#0:HttpContext.Current|};
+        }
+    }";
+            var fixtest = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public object Test(HttpContext currentContext) => currentContext;
+        }
+    }";
+
+            var expected = VerifyCS.Diagnostic(DiagnosticId).WithLocation(0).WithArguments(HttpContextName, HttpContextCurrentName);
+            await CreateTest().WithSource(test).WithFixed(fixtest).WithExpectedDiagnostics(expected).RunAsync();
+        }
+
+        [Fact]
+        public async Task ReuseArgument()
+        {
+            var test = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public void Test(HttpContext currentContext)
+            {
+                _ = {|#0:HttpContext.Current|};
+            }
+        }
+    }";
+            var fixtest = @"
+    using System.Web;
+
+    namespace ConsoleApplication1
+    {
+        class Program
+        {
+            public void Test(HttpContext currentContext)
```

Comment:
currentContext seems like a relatively common name.

is it easy to enumerate the params already on the method and determine if we should use currentContext+{instanceCount}?